MOBILE_CLIENT/COMMON/_MEDIUM/CRYPTO_INSECURE_CIPHER_ALGO/recommendation.md

```diff
@@ -1,7 +1,189 @@
+
+# Code Examples:
+
+### Dart
+
+```dart
+import 'dart:convert';
+import 'dart:typed_data';
+import 'package:flutter/material.dart';
+import 'package:flutter/services.dart';
+import 'package:crypto/crypto.dart';
+import 'package:pointycastle/export.dart' as pc;
+
+void main() {
+  runApp(MyApp());
+}
+
+class MyApp extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return MaterialApp(
+      home: Scaffold(
+        appBar: AppBar(
+          title: Text('Secure Encryption Demo'),
+        ),
+        body: EncryptionWidget(),
+      ),
+    );
+  }
+}
+
+class EncryptionWidget extends StatefulWidget {
+  @override
+  _EncryptionWidgetState createState() => _EncryptionWidgetState();
+}
+
+class _EncryptionWidgetState extends State<EncryptionWidget> {
+  final _controller = TextEditingController();
+  String _encryptedText = '';
+
+  void _encryptText() {
+    final key = utf8.encode('securekey123456789012345678901234'); // 32 bytes for AES-256
+    final iv = Uint8List(16); // 16 bytes for AES
+    final s = pc.SICStreamCipher(pc.AESFastEngine())
+      ..init(true, pc.ParametersWithIV(pc.KeyParameter(key), iv));
+    final input = utf8.encode(_controller.text);
+    final output = s.process(input);
+    setState(() {
+      _encryptedText = base64.encode(output);
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: <Widget>[
+        TextField(
+          controller: _controller,
+          decoration: InputDecoration(
+            labelText: 'Enter text to encrypt',
+          ),
+        ),
+        RaisedButton(
+          onPressed: _encryptText,
+          child: Text('Encrypt'),
+        ),
+        Text('Encrypted text: $_encryptedText'),
+      ],
+    );
+  }
+}
+```
+
+### Swift
+
+```swift
+import Foundation
+import CommonCrypto
+
+func AES_Encrypt(input: String, key: String) -> String {
+    let data = input.data(using: String.Encoding.utf8)!
+    let keyData = key.data(using: String.Encoding.utf8)!
+    let keyBytes = keyData.withUnsafeBytes { (bytes: UnsafePointer<UInt8>) -> UnsafePointer<UInt8> in
+        return bytes
+    }
+    let dataLength = Int(data.count)
+    let buffer = UnsafeMutablePointer<UInt8>.allocate(capacity: dataLength + kCCBlockSizeAES128)
+    let bufferPtr = UnsafeMutableRawPointer(buffer)
+    let bufferPtrBytes = bufferPtr.bindMemory(to: Void.self, capacity: dataLength)
+    let iv = [UInt8](repeating: 0, count: kCCBlockSizeAES128)
+    var numBytesEncrypted :size_t = 0
+    let cryptStatus = CCCrypt(CCOperation(kCCEncrypt), CCAlgorithm(kCCAlgorithmAES), CCOptions(kCCOptionPKCS7Padding), keyBytes, kCCKeySizeAES128, iv, data.bytes, dataLength, bufferPtrBytes, dataLength + kCCBlockSizeAES128, &numBytesEncrypted)
```

Comment:
Prefer a library that supports authenticated mode in AES. CCB is not what we recommend.